packages/rich-text/src/plugins/Hyperlink/HyperlinkModal.tsx

```diff
@@ -153,36 +161,33 @@ export function HyperlinkModal(props: HyperlinkModalProps) {
                   Link target{' '}
                 </FormLabel>
 
-                {linkEntity && (
-                  <TextLink onClick={resetLinkEntity} className={styles.removeSelectionLabel}>
-                    Remove selection
-                  </TextLink>
-                )}
-
-                {linkEntity && (
-                  <div>
-                    {linkType === INLINES.ENTRY_HYPERLINK && (
-                      <FetchingWrappedEntryCard
-                        sdk={props.sdk}
-                        locale={props.sdk.field.locale}
-                        entryId={linkEntity.sys.id}
-                        isDisabled={true}
-                        isSelected={false}
-                      />
-                    )}
-                    {linkType === INLINES.ASSET_HYPERLINK && (
-                      <FetchingWrappedAssetCard
-                        sdk={props.sdk}
-                        locale={props.sdk.field.locale}
-                        assetId={linkEntity.sys.id}
-                        isDisabled={true}
-                        isSelected={false}
-                      />
-                    )}
-                  </div>
-                )}
-
-                {!linkEntity && (
+                {linkEntity && linkEntity.sys.linkType === SYS_LINK_TYPES[linkType] ? (
```

Comment:
This ternary has been adjusted so that the asset -> entry -> asset switching logic in the modal "just works." Before, it would simply see that there was a `linkEntity` when updating and try to render the entity regardless of whether an entry or an asset was being linked. This would result in the "entry missing or unaccessible display."